src/Core/Domain/Product/Pack/Query/GetPackedProductsDetails.php

```diff
@@ -28,24 +28,32 @@
 
 namespace PrestaShop\PrestaShop\Core\Domain\Product\Pack\Query;
 
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
 use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
 
 /**
  * Retrieves product from a pack
  */
-class GetPackedProducts
+class GetPackedProductsDetails
 {
     /**
      * @var ProductId
      */
     private $packId;
 
     /**
-     * @param int $packId id of product which represents the pack
+     * @var LanguageId
      */
-    public function __construct(int $packId)
+    private $languageId;
+
+    /**
+     * @param ProductId $packId
+     * @param LanguageId $languageId
+     */
+    public function __construct(ProductId $packId, LanguageId $languageId)
```

Comment:
We always use primitive types as input (unless a very special case that requires using a VO), so you should use `int` type here However, they are then converted into VO objects in the constructor

Reply:
ok